pkg/controller/deactivation/deactivation_controller_test.go

```diff
@@ -0,0 +1,421 @@
+package deactivation
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	tiertest "github.com/codeready-toolchain/host-operator/test/nstemplatetier"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	murtest "github.com/codeready-toolchain/toolchain-common/pkg/test/masteruserrecord"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+const (
+	operatorNamespace = "toolchain-host-operator"
+)
+
+func TestReconcile(t *testing.T) {
+
+	// given
+	logf.SetLogger(logf.ZapLogger(true))
+	username := "test-user"
+
+	t.Run("controller should not deactivate user", func(t *testing.T) {
+
+		// the time since the mur was provisioned is within the deactivation timeout period for the 'basic' tier
+		t.Run("usersignup should not be deactivated - basic tier (30 days)", func(t *testing.T) {
+			// given
+			expectedDeactivationTimeout := 30
+			userSignup := &toolchainv1alpha1.UserSignup{
+				ObjectMeta: newObjectMeta(username, "foo@bar.com"),
+				Spec: toolchainv1alpha1.UserSignupSpec{
+					Username:      "foo@bar.com",
+					Approved:      true,
+					TargetCluster: "east",
+				},
+			}
+			basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates, tiertest.WithCurrentUpdateInProgress())
+			murProvisionedTime := &metav1.Time{Time: time.Now()}
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *basicTier), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignup))
+			initObjs := []runtime.Object{basicTier, mur, userSignup}
+			r, req, cl := prepareReconcile(t, mur.Name, initObjs...)
```

Comment:
why do we need to store it into an array? we can use it without it...
```suggestion
			r, req, cl := prepareReconcile(t, mur.Name, basicTier, mur, userSignup)
```

Reply:
applies for all other tests

Reply:
Good point, fixed in aa87f8eda85f8c5c07982f59f310afe925a115bd